### PR TITLE
Added nox support. Fixes #10

### LINKED ISSUE #10: Integrate nox for flexible test automation
https://nox.thea.codes/ allows to create flexible test automation. It would be good to adopt it.

### PR DESCRIPTION
**Pull request by amolkahat**
Added nox support.
Signed-off-by: Amol Kahat <amolkahat@gmail.com>

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -43,10 +43,8 @@ jobs:
...
-        run: |
-          python -m pip install .
-          python -m pip install -Ur dev-requirements.txt
+      - name: Install nox
+        run: python -m pip install nox
 
-      - name: Unit Tests
-        run: py.test tests -v
+      - name: Run nox
+        run: nox
```
Comment (digitronik) [inline .github/workflows/main.yml]: I would not make it as part of `CI`. why?
`nox` is just like `tox` which help to test package on a different `version of python`. Our CI already has taken care of supporting python versions and different OS.

File: noxfile.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import nox
```
Comment (digitronik) [inline noxfile.py]: For local support; we can make `nox` as part of `dev-requirements.txt`

Comment (raukadah) [reply noxfile.py]: @amolkahat We also need updates to README file

File: .github/workflows/main.yml (hunk 1 above)
Comment (amolkahat) [reply .github/workflows/main.yml]: I added this to simplify setup and dependencies.
Currently only tests environment is added. If you want to install it we can write more code which can run with only one command like:
```
nox --session tests
nox --session development
nox --session user
```
[source](https://nox.thea.codes/en/stable/tutorial.html)

Comment (digitronik) [reply .github/workflows/main.yml]: @amolkahat I like your proposal, do some changes.

1. Restore workflow as it is.
2. add `nox` dependency in `dev-requirements.txt`
3. Let utilize nox features to test PR changes locally (not in CI); I mean parametrize over supported python versions like
```
@nox.session(python=[ "3.6", "3.7"])
def test(session):
    pass
```

Comment (amolkahat) [reply .github/workflows/main.yml]: @digitronik why not in CI?

**Pull request merged**